### PR TITLE
feat: add JUnit XML output format support (fixes #2)

### LINKED ISSUE #2: [Enhancement] Add JUnit XML output format for test results
## Summary

Add support for JUnit XML output format to enable seamless integration with CI/CD systems, test reporting tools, and other automation platforms that expect standardized test result formats.

## Background

Currently, mcp-server-tester supports console output (rich formatting) and has a stub for JSON output. While these formats work well for human consumption and basic automation, many CI/CD systems (Jenkins, GitLab, GitHub Actions, etc.) and test management tools expect JUnit XML format for:

- Automated test result parsing and visualization
- Historical trending and analytics  
- Integration with reporting dashboards
- Standardized test failure notifications

### Core Functionality
- **Support all test types**: Both `tools` and `evals` test results should be exported to JUnit XML
- **Standard compliance**: Follow widely-adopted JUnit XML schema conventions (referencing [testmoapp/junitxml](https://github.com/testmoapp/junitxml) as the de facto standard)
- **Extensible architecture**: Integrate with existing `DisplayManager`/`TestFormatter` system

#### JUnit XML Structure Requirements
```xml
<testsuites>
  <testsuite name="tools" tests="5" failures="1" errors="0" time="2.34">
    <testcase classname="filesystem" name="read_file" time="0.45"/>
    <testcase classname="filesystem" name="write_file" time="0.67">
      <failure message="Permission denied" type="AssertionError">
        Expected: file written successfully
        Actual: Permission denied
      </failure>
    </testcase>
  </testsuite>
  <testsuite name="evals" tests="3" failures="0" errors="0" time="15.67">
    <testcase classname="claude-3-sonnet" name="file_operations_eval" time="8.23"/>
  </testsuite>
</testsuites>
```

#### CLI Integration
- Add `--junit-xml=<filename>` parameter to specify JUnit XML output file
- Support both `--junit-xml=test-results.xml` and `--junit-xml test-results.xml` syntax  
- Default to `junit.xml` in current directory if no filename specified: `--junit-xml`
- Maintain backward compatibility with existing output options

### Implementation Approach

1. **Extend DisplayManager Architecture**: Create `JunitXmlFormatter` implementing `TestFormatter` interface
2. **Map Test Types**: 
   - Tools tests → individual `<testcase>` elements grouped by capability/tool
   - Evals tests → `<testcase>` elements grouped by model and test scenario
3. **Error Handling**: Map different failure types (`TestResult.error`, eval scoring failures) to appropriate JUnit elements (`<failure>`, `<error>`, `<skipped>`)
4. **Timing Information**: Include execution duration at both suite and case levels
5. **Metadata**: Capture relevant properties (model names, server configs, etc.) in `<properties>` sections

## Validation Requirements

To ensure we're "not flying blind" with JUnit XML output:

1. **Schema Validation**: Implement validation against established JUnit XML schema
2. **Integration Testing**: Test output with common CI tools (Jenkins, GitLab CI)
3. **Recommended Tools**:
   - **XMLUnit** for programmatic validation in tests
   - **testmoapp/junitxml** schema as reference standard
   - CI system validation (GitHub Actions, GitLab CI pipelines)

## Acceptance Criteria

- [ ] JUnit XML formatter integrates with existing `DisplayManager` system
- [ ] Both `tools` and `evals` test results export correctly to JUnit XML
- [ ] Generated XML validates against standard JUnit schema
- [ ] CLI accepts `--junit-xml` parameter with flexible filename options
- [ ] Documentation updated with JUnit output examples
- [ ] Integration tests verify CI tool compatibility
- [ ] Backward compatibility maintained for existing output formats

### Related Files
- `src/testing/display/DisplayManager.ts` - Main display coordination
- `src/testing/display/formatters/` - Existing formatter implementations  
- `src/testing/display/formatters/JsonFormatter.ts` - Currently unimplemented, could be completed alongside this
- `src/core/types.ts` - Test result type definitions

### Standards References
- [testmoapp/junitxml](https://github.com/testmoapp/junitxml) - JUnit XML format documentation and examples
- [XMLUnit](https://www.xmlunit.org/) - XML validation library for testing

This enhancement will significantly improve mcp-server-tester's integration capabilities with modern CI/CD workflows while leveraging the existing robust testing architecture.

### PR DESCRIPTION
**Pull request by steviec**
## Summary

Adds comprehensive JUnit XML output format support to enable seamless integration with CI/CD systems, test reporting tools, and automation platforms.

## Changes

- **New JunitXmlFormatter**: Implements `TestFormatter` interface with full JUnit XML compliance
- **Enhanced DisplayManager**: Support for multiple concurrent formatters (console + JUnit XML)
- **CLI Integration**: New `--junit-xml [filename]` parameter with flexible filename options
- **XML Validation**: Built-in validation utilities to ensure schema compliance
- **Comprehensive Testing**: Full test coverage for JUnit formatter and validation utilities

### JUnit XML Structure
- Standard-compliant XML format following testmoapp/junitxml conventions
- Support for both `tools` and `evals` test types
- Proper error handling with detailed failure information
- Timing data at both suite and test case levels
- Model metadata integration for eval tests

### CLI Usage
```bash
# Generate JUnit XML with default filename
mcp-server-tester test.yaml --server-config server.json --junit-xml

# Specify custom filename
mcp-server-tester test.yaml --server-config server.json --junit-xml=custom-results.xml

# Combined with other options
mcp-server-tester test.yaml --server-config server.json --junit-xml --verbose --quiet
```

### Technical Implementation
- **Multi-formatter architecture**: Console and JUnit XML formatters run concurrently
- **Event-driven design**: Leverages existing TestEvent system for consistent data flow
- **XML safety**: Proper escaping of special characters and validation
- **Flexible output**: Automatic directory creation and configurable file paths

## Test Coverage

- Unit tests for JunitXmlFormatter with comprehensive scenarios
- XML validation testing
- CLI integration verification
- Error handling and edge cases

## Compatibility

- ✅ Backward compatible with existing CLI options
- ✅ Works with both tools and evals test types
- ✅ Integrates with existing DisplayManager architecture
- ✅ No breaking changes to current functionality

Fixes #2

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: test/unit/junit-xml-formatter.test.ts [hunk 1]
```diff
@@ -0,0 +1,444 @@
+/**
+ * Tests for JUnit XML formatter
+ */
+
+import { describe, test, expect, beforeEach, afterEach } from 'vitest';
+import { JunitXmlFormatter } from '../../src/testing/display/formatters/JunitXmlFormatter.js';
+import {
+  validateJunitXmlContent,
+  validateJunitXml,
+} from '../../src/testing/display/formatters/junit-validation.js';
```
Comment (steviec) [inline test/unit/junit-xml-formatter.test.ts:10]: We should NOT be rolling our own Junit validation. We should use an existing library for doing that, and add it to our dev dependencies.

File: test/unit/junit-xml-formatter.test.ts [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+      const content = readFileSync(outputFile, 'utf8');
+      const validation = validateJunitXmlContent(content);
+
+      expect(validation.valid).toBe(true);
+      expect(content).toContain('<?xml version="1.0" encoding="UTF-8"?>');
+      expect(content).toContain('<testsuites');
+      expect(content).toContain('</testsuites>');
+    });
+
+    test('should generate valid XML for tools test suite', () => {
```
Comment (steviec) [inline test/unit/junit-xml-formatter.test.ts:64]: We should rely on another library for determining if the file is "correct" junit (if it's cleaner to just use a junit schema and running our junit output against it that's fine). We should then only focus on ensuring that certain outputs are captured in the junit.

**Review by steviec — COMMENTED** [review]
Needs simplification of Junit tests.

**Review by steviec — COMMENTED** [review]
Please fix this stuff.
